### ISSUE TITLE
Quota validation fails when the number of tasks exceeds the quota 

### ISSUE THREAD

**Issue by tstannard** [cloudfoundry/cloud_controller_ng#2283]
## Issue

If the exact value of the tasks exceeds the allowable amount of tasks, the limit will not be enforced. This means if the validation is not checked exactly when the number of tasks is equal to the quota, the quota will be ignored. 

## Context

During a repave, there was a rehosting of applications that resulted in a bulk generation of tasks. This causes the number of active tasks to exceeded the quota. However, when the quota was validated it only checked for the number of tasks to be equal to the quota. In this case, the quota value and the number of tasks were not equal, BUT the quota should have been enforced since the limit was exceeded. 

## Expected result

Tasks should fail due to a quota limit but the cloud controller should stay healthy.  

## Current result

The quota is ignored, all tasks start and take down the cloud controller.  

## Possible Fix

Update the quota validation from equality to a less than inequality.

**Comment by tstannard** [comment]
This issue was addressed with this commit: [b6d6dc](https://github.com/cloudfoundry/cloud_controller_ng/commit/b6d6dc36de7a1ddedd3d15eacb1b04d7ada2516a)

**Comment by sweinstein22** [comment]
### ✅   Acceptance

Set up an organization with no task quota:
```
cf curl /v3/organization_quotas/954e2b39-bd1c-4582-8277-d1f7c22ca6f9
{
  "guid": "954e2b39-bd1c-4582-8277-d1f7c22ca6f9",
  "name": "task-quota",
  "apps": {
    "total_memory_in_mb": null,
    "per_process_memory_in_mb": null,
    "total_instances": null,
    "per_app_tasks": null
  },
  ...
}
```
Ran three tasks against app `task-dora` with command `sleep 500000`:
```
cf tasks task-dora
Getting tasks for app task-dora in org task-org / space task-space as admin...

id   name       state     start time                      command
3    49261552   RUNNING   Mon, 24 May 2021 18:39:35 UTC   sleep 500000
2    2c4730d4   RUNNING   Mon, 24 May 2021 18:39:30 UTC   sleep 500000
1    ca6ee518   RUNNING   Mon, 24 May 2021 18:39:15 UTC   sleep 500000
```
Set `per_app_tasks` to `2`, lower than the number of running tasks:
```
cf curl -X PATCH /v3/organization_quotas/954e2b39-bd1c-4582-8277-d1f7c22ca6f9 -d '{"apps": {"per_app_tasks":2}'
```
Tried to create another task, saw error:
```
cf run-task task-dora -c 'sleep 500000'
Creating task for app task-dora in org task-org / space task-space as admin...
app_task_limit quota exceeded
FAILED
```

### ❌   Rejection

This does not seem to have been fixed for space-level quotas. Ran through the same steps as above but with a space-level quota and was able to create an additional task

```
cf curl /v3/space_quotas/5a18905b-d479-4c9d-92bb-2c0aba9da739
{
  "guid": "5a18905b-d479-4c9d-92bb-2c0aba9da739",
  "name": "task-space-quota",
  "apps": {
    "total_memory_in_mb": null,
    "per_process_memory_in_mb": null,
    "total_instances": null,
    "per_app_tasks": 2
  },
  ....
}
```
```
cf run-task task-dora -c 'sleep 500000'
Creating task for app task-dora in org task-org / space task-space as admin...
Task has been submitted successfully for execution.
OK

task name:   b5811961
task id:     4
```
Looking at the code change in https://github.com/cloudfoundry/cloud_controller_ng/commit/b6d6dc36de7a1ddedd3d15eacb1b04d7ada2516a it seems like this tracks, as changes were only made at the org quota validation level

**Comment by tjvman** [comment]
# Acceptance
**Confirm that org-level behavior still works** ✅
```
$ cf tasks task-dora  # 4 tasks already running
Getting tasks for app task-dora in org task-org / space task-space as admin...

id   name       state     start time                      command
4    b5811961   RUNNING   Mon, 24 May 2021 18:46:38 UTC   sleep 500000
3    49261552   RUNNING   Mon, 24 May 2021 18:39:35 UTC   sleep 500000
2    2c4730d4   RUNNING   Mon, 24 May 2021 18:39:30 UTC   sleep 500000
1    ca6ee518   RUNNING   Mon, 24 May 2021 18:39:15 UTC   sleep 500000

$ cf curl -X PATCH /v3/organization_quotas/954e2b39-bd1c-4582-8277-d1f7c22ca6f9 -d '{"apps": {"per_app_tasks":5}' # bump org quota to allow 5 tasks
{
  ...
    "apps": {
      "total_memory_in_mb": null,
      "per_process_memory_in_mb": null,
      "total_instances": null,
      "per_app_tasks": 5
    },
  ...
}


$ cf curl -X PATCH /v3/space_quotas/5a18905b-d479-4c9d-92bb-2c0aba9da739 -d '{"apps": {"per_app_tasks":6}' # bump space quota to allow 6 tasks
{
  ...
    "apps": {
      "total_memory_in_mb": null,
      "per_process_memory_in_mb": null,
      "total_instances": null,
      "per_app_tasks": 6
    },
  ...
}

$ seq 1 3 | xargs -n1 -I{} -P 16 cf run-task task-dora -c 'sleep 10000' # org quota should allow one more task in
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
app_task_limit quota exceeded
FAILED
app_task_limit quota exceeded
FAILED
Task has been submitted successfully for execution.
OK
task name:   e189af07
task id:     5

$ seq 1 3 | xargs -n1 -I{} -P 16 cf run-task task-dora -c 'sleep 10000' # no more tasks should be allowed
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
app_task_limit quota exceeded
FAILED
app_task_limit quota exceeded
FAILED
app_task_limit quota exceeded
FAILED
```

**Confirm space-level quota works as intended** ✅
```
$ cf curl -X PATCH /v3/organization_quotas/954e2b39-bd1c-4582-8277-d1f7c22ca6f9 -d '{"apps": {"per_app_tasks":7}' # set org quota higher than space quota
{
  ...
    "apps": {
      "total_memory_in_mb": null,
      "per_process_memory_in_mb": null,
      "total_instances": null,
      "per_app_tasks": 7
    },
  ...
}

$ seq 1 3 | xargs -n1 -I{} -P 16 cf run-task task-dora -c 'sleep 10000' # space quota should allow 1 more task in, for a total of 6.
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
Creating task for app task-dora in org task-org / space task-space as admin...
app_task_limit quota exceeded
FAILED
app_task_limit quota exceeded
FAILED
Task has been submitted successfully for execution.
OK

task name:   231e597d
task id:     6
```

**Comment by tjvman** [comment]
Fix shipped in [capi-release 1.111.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.111.0)
